### ISSUE TITLE
Filters for string properties

### ISSUE THREAD

**Issue by user472** [oramasearch/orama#308]
**Describe the solution you'd like**
In some situations I would like to search only a subset of all documents. This subset is defined by a string property `category` for this example. So the query I would like to execute is the following:

```
schema: 
{
  name: 'string',
  description: 'string',
  category: 'string'
}

search(db, {
  term: 'Harry',
  properties: '*',
  where: {
    category: 'movies'
  }
})
```

**Describe alternatives you've considered**
A workaround could be to assign categories to number (movies = category 1) but this is kind of annoying since I need to separately need to keep track of the mapping from numbers to category names and it is an additional field per string I want to filter by.

**Comment by micheleriva** [comment]
Hi @user472!
True, we definitely need this in the next release (cc. @ShogunPanda)

**Comment by ShogunPanda** [comment]
Agreed!
I see several ways to do so.
I have a question tho: are you thinking about having the initial subset stable (in the sense, looking at the example above, that you know you'll always subset by category first) or not?

**Comment by micheleriva** [comment]
The reason why I haven't included this in the past release is that we need to define which kind of filter we want to apply.

Running stuff like `{ where: { category: { eq: 'movies' } } }` is fine, but if we want to add more options (such as `IN`, `LIKE`, etc), we will need a totally different approach and data structures, which will increase the index size a lot.

As for now, I'd discourage this, and I'd add just the equality comparison operator.

**Comment by ShogunPanda** [comment]
I agree on that.

**Comment by user472** [comment]
Hi @ShogunPanda
It is not intended to be necessarily stable. The other use case I have is to allow a search over all categories.

**Comment by nxf-gerdjungbluth** [comment]
I think this has been resolved by #350, hasn't it?
Which is great - by the way - since we just started using orama and built or own workaround this week 👍

**Comment by micheleriva** [comment]
Hi @user472 and @nxf-gerdjungbluth,
yes this has been released with Orama v1.0.0-beta.10. Find the full docs here: https://docs.oramasearch.com/usage/search/filters

Closing this as implemented. Cheers!
